### PR TITLE
fix: UseTitleForAutoMerge use correct title and number

### PR DESCRIPTION
**Pull request by jetersen**
# Title

Ran a test this time here: https://github.com/jetersen-cloud/updatecli-test/commit/22fda8327ce277b2721de0bca0ba8b8a0d6745b6

Sorry should have tested it.

## Test

https://github.com/jetersen-cloud/updatecli-test/commit/22fda8327ce277b2721de0bca0ba8b8a0d6745b6

## Additional Information

fixes #782 I were using incorrect title and number 😓

### CODE REVIEW

File: pkg/plugins/scms/github/pullrequest.go [hunk 1]
```diff
@@ -330,9 +330,9 @@ func (p *PullRequest) EnablePullRequestAutoMerge() error {
 
 	if p.spec.UseTitleForAutoMerge {
 		if strings.EqualFold(p.spec.MergeMethod, "squash") {
-			input.CommitHeadline = githubv4.NewString(githubv4.String(fmt.Sprintf("%s (#%s)", p.spec.Title, p.remotePullRequest.ID)))
+			input.CommitHeadline = githubv4.NewString(githubv4.String(fmt.Sprintf("%s (#%d)", p.Title, p.remotePullRequest.Number)))
```
Comment (jetersen) [inline pkg/plugins/scms/github/pullrequest.go:333]: Something odd is going on however.

`p.spec.Title` being empty... 😅 yet `p.spec.UseTitleForAutoMerge` being correctly set to true.

Comment (jetersen) [reply pkg/plugins/scms/github/pullrequest.go:333]: nvm, it makes sense. Not sure what title is used for in the spec though when it is on the main pull request target.

Comment (jetersen) [reply pkg/plugins/scms/github/pullrequest.go:333]: The question is whether this is a bug or not.

I don't know the history why we have both `p.Title` and `p.Spec.Title`
However looking at the code `p.spec.Title` is only used inside: `GetChangelogTitle`

Not sure what is meant with that seperation 🤔

I guess by `p.Title` I have complete control and where `p.Spec.Title` would prepend `[updatecli]`

Not sure why the manifest allows for both flows.

**Review by olblak — APPROVED** [review]

File: pkg/plugins/scms/github/pullrequest.go (hunk 1 above)
Comment (jetersen) [reply pkg/plugins/scms/github/pullrequest.go:333]: This works
```yaml
pullrequests:
  jenkins:
    title: Bump jenkins.version from 2.319.3 to {{ source "jenkins" }} for bom-2.319.x
    kind: github
    scmid: github
    targets:
    - jenkins
    spec:
      labels:
      - dependencies
      automerge: true
      mergemethod: squash
      usetitleforautomerge: true
```

Also works but prepends `[updatecli]`
```
pullrequests:
  jenkins:
    kind: github
    scmid: github
    targets:
    - jenkins
    spec:
      title: Bump jenkins.version from 2.319.3 to {{ source "jenkins" }} for bom-2.319.x
      labels:
      - dependencies
      automerge: true
      mergemethod: squash
      usetitleforautomerge: true
```

**Comment by olblak** [comment]
@jetersen I let you run the final testing on the main branch then you can trigger the release `v0.29.0` from "draft" to "pre-release" once. At the end of the release process it will convert from pre-release to release

**Comment by jetersen** [comment]
@olblak sweet :)

**Comment by jetersen** [comment]
Tested and works from main branch here: https://github.com/jetersen-cloud/updatecli-test/commit/fdb6961bb18dec11bf2140e6d76d799f33965283

**Pull request merged**